File: src/main/java/org/shingas/viewerSMP/utils/MMParser.java
package org.shingas.viewerSMP.utils;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver;

public class MMParser {

    private static final MiniMessage mm = MiniMessage.miniMessage();

    public static Component getMsg(String msg) {
        return mm.deserialize(msg);
    }

    public static Component getMsgTags(String msg, TagResolver tags) {
        return mm.deserialize(msg, tags);
    }

}

File: src/main/kotlin/com/shingaspt/viewerSMP/utils/MMParser.kt
package com.shingaspt.viewerSMP.utils

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.minimessage.MiniMessage
import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver

class MMParser {

    companion object {

        private val mm: MiniMessage = MiniMessage.miniMessage()

        fun getMsg(msg: String) : Component {
            return mm.deserialize(msg)
        }

        fun getMsgTags(msg: String, tags: TagResolver): Component {
            return mm.deserialize(msg, tags)
        }
    }

}